encrypted-media-respec.html

```diff
@@ -605,8 +605,112 @@
+              algorithm in order to avoid repeated requests to the user for similar configurations.
+              </p>
+            </div>
+            <ol>
+              <li>
+                <p>Let <var>requirements</var> be the list [ <code>distinctiveIdentifier</code>, <code>persistentState</code> ].</p>
+              </li>
+              <li>
+                <p>
+                  Let <var>optional requirements</var> be the sublist of <var>requirements</var> where for each element of the sublist,
+                  the value of the corresponding member of <var>candidate configuration</var> is <a def-id="requirement-optional"></a>.
+                </p>
+              </li>
+              <li>
+                <dl class="switch">
+                  <dt>If <var>optional requirements</var> is not empty:</dt>
+                  <dd>
+                    <ol>
+                      <li>
+                        <p>
+                          Let <var>optional requirement combinations</var> be the list of all sublists of <var>optional requirements</var>
+                          ordered by length (longest first) and then by the position in <var>optional requirements</var> of the first element that differs
+                          between two lists (lowest first). (i.e. [ <code>distinctiveIdentifier</code> ] is before [ <code>persistentState</code> ].)
+                        </p>
+                      </li>
+                      <li>
+                        <p>For each <var>optional requirement combination</var> in <var>optional requirement combinations</var>:<p>
+                        <ol>
+                          <li>
+                            <p>
+                            Let <var>modified candidate configuration</var> be a copy of
+                            <var>candidate configuration</var>.
+                            </p>
+                          </li>
+                          <li>
+                            <p>
+                              For each element of <var>optional requirements</var> not in <var>optional requirement combination</var>, set the
+                              corresponding member of <var>modified candidate configuration</var> to <a def-id="requirement-not-allowed"></a>.
+                            </p>
+                          </li>
+                          <li>
+                            <p>
+                            Let <var>supported configuration</var> be the result of executing the
+                            <a def-id="get-supported-configuration-and-consent-algorithm"></a> algorithm
+                            with <var>implementation</var>, <var>modified candidate configuration</var> and <var>origin</var>.
+                            </p>
+                          </li>
+                          <li>
+                            <p>
+                            If <var>supported configuration</var> is not <code>ConsentDenied</code>,
```

Comment:
In theory, the user agent could have made choices that require two "optional" `MediaKeysRequirement` items to be "required". If consent was denied for reasons that aren't the first item and "not-allowed" is not supported for that item, this could break. Yes, this is unlikely, especially given these two items, but it seems that if the algorithm is going to be very general, it should not make such assumptions.

An example:
>Trying ["optional", "optional"] => becomes ["required", "required"] => `ConsentDenied` => ... trying ["not-allowed", "optional"] => `NotSupported`

In theory, this could continue:
>=> trying ["optional", "not-allowed"] => becomes ["required", "not-allowed"] => Supported